src/core_modules/capture-core/components/WidgetEnrollment/Actions/AddNew/AddNew.component.js

```diff
@@ -0,0 +1,31 @@
+// @flow
+import { IconAdd16, MenuItem, Tooltip } from '@dhis2/ui';
+import React from 'react';
+import i18n from '@dhis2/d2-i18n';
+import type { Props } from './addNew.types';
+import { getTrackedEntityTypeThrowIfNotFound } from '../../../../metaData';
+
+const AddNewItem = ({ onClick, ...passOnProps }: Object) => (<MenuItem
+    dense
+    dataTest="widget-enrollment-actions-add-new"
+    onClick={onClick}
+    icon={<IconAdd16 />}
+    label={i18n.t('Add new')}
+    {...passOnProps}
+/>);
+
+export const AddNew = ({ canAddNew, onlyEnrollOnce, tetId, onAddNew }: Props) => {
+    if (!canAddNew) {
+        return null;
+    }
+    const tetName = tetId ? getTrackedEntityTypeThrowIfNotFound(tetId)?.name : '';
```

Comment:
To keep this self-contained, we should get the TET name in src/core_modules/capture-core/components/WidgetEnrollment/hooks/useProgram.js and use that one (add `trackedEntityType[displayName]` to the fields)